src/components/SignIn.js

```diff
@@ -34,9 +34,12 @@ class SignIn extends Component {
         ).then(response => {
             // Check if authentication was successful
             if (response.ok) {
-                // Trigger a page transition in the parent component
-                this.props.onSignIn(false, this.state.username);
-                this.setState({ isSignedIn: true });
+                response.json().then(data => {
+                    localStorage.setItem('bounceJwt', data.token);
+                    // Trigger a page transition in the parent component
+                    this.props.onSignIn(false, this.state.username);
+                    this.setState({isSignedIn: true});
+                });
```

Comment:
I think it would be better to just handle the token in the client rather than doing this in every different place we call `authenticate`. Example (in `api.js`):

```javascript
    async authenticate(username, password) {
        const response = await this._request('POST', '/auth/login', {
            username: username,
            password: password,
        });
        if (response.ok) {
            this.token = (await response.json())['token'];
            localStorage.setItem('bounceJwt', this.token);
        }
        return Promise.resolve(response);
    }
```